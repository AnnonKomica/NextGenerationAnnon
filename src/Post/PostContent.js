import React from "react";
import LineElement from "../LineElement";

type Props = {
    children: string,
};

export default class PostContent extends React.Component {

    props: Props;

    render() {
        const {children, postLineEventHandlers} = this.props;
        const lines: string[] = children.split("\n");

        return (
            <div>
                {lines.map((line, currentIndex) => {
                    const element = <LineElement postLineEventHandlers={postLineEventHandlers}>{line}</LineElement>;

                    if (!currentIndex) {
                        return element;
                    }

                    return [<br key="break" />, element];
                })}
            </div>
        );
    }
}
