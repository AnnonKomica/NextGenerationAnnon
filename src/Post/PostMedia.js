// @flow
import React from "react";
import {ImageMediaElement, WebMMediaElement} from "../ThumbnailMediaElement";

type Props = {
    mediaSrc: ?string
};

type State = {
    isLoad: boolean,
};

const JPEG_REG_EXR = /.jpg$/;
const GIF_REG_EXR = /.gif$/;
const WEBM_REG_EXR = /.webm$/;

export default class PostMedia extends React.Component {

    props: Props;
    state: State;

    render() {
        const {mediaSrc} = this.props;

        if (!mediaSrc) {
            return null;
        }

        if (mediaSrc.match(JPEG_REG_EXR) || mediaSrc.match(GIF_REG_EXR)) {
            return (
                <ImageMediaElement
                    thumbnailSrc={mediaSrc}
                    rawMediaSrc={mediaSrc}
                />
            );
        }

        if (mediaSrc.match(WEBM_REG_EXR)) {
            return (
                <WebMMediaElement
                    thumbnailSrc={mediaSrc}
                    rawMediaSrc={mediaSrc}
                />
            );
        }

        // Unsupport image format
        return null;
    }
}
