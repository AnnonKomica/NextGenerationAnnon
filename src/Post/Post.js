// @flow
import React from "react";
import shallowCompare from "react-addons-shallow-compare";
import PostMedia from "./PostMedia";
import PostContent from "./PostContent";

type Props = {
    threadId: string,
    imgSrc?: string,
    nickName?: string,
    title?: string,
    userId: string,
    postNumber: string,
    content: string,
    timestamp: string,
}

type State = {};

const DefaultValue = {
    TITLE: "無題",
    NAME: "無名"
};

export default class Post extends React.Component {

    props: Props;
    state: State;

    shouldComponentUpdate(nextProps: Props, nextState: State) {
        return shallowCompare(this, nextProps, nextState);
    }

    render() {
        const {
            imgSrc,
            title,
            nickName,
            userId: generatedUserId,
            content,
            postNumber,
            timestamp,
        } = this.props;

        return (
            <div id={`${postNumber}`} className="post">
                <div>
                    <span>{`${title || DefaultValue.TITLE} `}</span>
                    <span>{`${nickName || DefaultValue.NAME} `}</span>
                    <span>{`${new Date(timestamp).toLocaleDateString()} `}</span>
                    <span>{`${postNumber} `}</span>
                    <span>{`${generatedUserId}`}</span>
                </div>
                <PostMedia mediaSrc={imgSrc} />
                <PostContent>{content}</PostContent>
            </div>
        );
    }
}
