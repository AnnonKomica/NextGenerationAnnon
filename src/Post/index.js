// @flow
import Post from "./Post";

require("./sass/Post.scss");

export default Post;
