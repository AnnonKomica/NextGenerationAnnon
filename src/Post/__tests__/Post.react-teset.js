import React from "react";
import renderer from "react-test-renderer";

import Post from "../Post";

import POST from "./testData";

// eslint-disable-next-line
test("POST DOM should render correctly", () => {
    const component = renderer.create(
        <Post {...POST} />
    );

    let tree = component.toJSON();

    // eslint-disable-next-line
    expect(tree).toMatchSnapshot();
});
