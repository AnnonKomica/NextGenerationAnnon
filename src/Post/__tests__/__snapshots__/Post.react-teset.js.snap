exports[`test POST DOM should render correctly 1`] = `
<div
  className="post"
  id="thread1_post0">
  <div>
    Title: 無題 User: JKwelkf23jv At: 19911209
  </div>
  <p>
    content1
  </p>
</div>
`;
